https://issues.redhat.com/browse/ACM-23917 MCOA

dockerymick opened the pull request.

dockerymick committed:
Updates

dockerymick committed:
more small updates

dockerymick committed:
adding more changes

dockerymick commented on observability/obs_mcoa_intro.adoc:
```diff
@@ -0,0 +1,74 @@
+[#obs-mcoa-intro]
+= Multicluster Observability add-on
```
```suggestion
= Multicluster observability add-on
```

dockerymick committed:
Apply suggestions from code review

dockerymick commented on observability/obs_mcoa_intro.adoc:
```diff
@@ -0,0 +1,74 @@
+
+== Additional resources
+
+- enabling MCOA for metrics
```
```suggestion
- xref:../observability/obs_enable_mcoa.adoc#enable-mcoa[Enabling the multicluster observability add-on]
```

dockerymick committed:
Apply suggestion from @dockerymick

dockerymick committed:
update API descriptions

Removed unnecessary line breaks in API descriptions for clarity.

dockerymick committed:
Configuring MCOA APIs

dockerymick committed:
More updates

dockerymick commented on observability/obs_mcoa_intro.adoc:
```diff
@@ -0,0 +1,78 @@
+- `platform-metrics-alerts`, which is specific for custom alert rule metrics across all platforms
+- `platform-metrics-default`, which is specific for default metrics across all platfroms
+
+[mcoa-add-res]
```
```suggestion
[#mcoa-add-res]
```

dockerymick committed:
Apply suggestion from @dockerymick

dockerymick committed:
Update observability introduction documentation

Removed outdated sections and streamlined content.

dockerymick committed:
Update obs_mcoa_federate.adoc

dockerymick committed:
Add relabeling documentation to main.adoc

dockerymick committed:
Document relabeling default metrics for MCOA

Added documentation for relabeling default metrics in the multicluster observability add-on, including prerequisites and steps.

dockerymick committed:
Update links and comments in obs_mcoa_intro.adoc

dockerymick committed:
Adding link to intro

dockerymick committed:
Add documentation for exporting metrics to external endpoints

Document the process for exporting metrics to external endpoints using PrometheusAgent.

dockerymick committed:
Add export_metrics_ext_endpts.adoc to main.adoc

dockerymick committed:
Enhance observability documentation with cluster ID details

Updated documentation to clarify the use of cluster ID for managed clusters and added health status information for the multicluster observability add-on.

dockerymick committed:
Fix formatting of API descriptions in documentation

dockerymick committed:
Rename obs_mcoa-relabel.adoc to obs_mcoa_relabel.adoc

dockerymick committed:
Fix duplicate entry for RightSizingRecommendation

Removed duplicate mention of `RightSizingRecommendation` in observability section.

thibaultmg commented on observability/obs_enable_mcoa.adoc:
```diff
@@ -0,0 +1,54 @@
+
+Enable the multicluster observability add-on on your hub cluster to configure your metrics for your platform and user workloads. You are required to enable platform workloads and it is optional to enable user workloads.
+
+When you enable `platform` and `userWorkloads` specifications, the `MultiClusterObservability` operator removes the `endpoint-observability-operator` and `metrics-collector-deployment` resources, and deploys the `multicluster-observability-addon-manager` in the `open-cluster-management-observability` namespace.
```
Proposal:
When you enable `platform` and `userWorkloads` specifications, the `MultiClusterObservability` operator stops deploying the metrics collectors in the managed clusters and deploys the `multicluster-observability-addon-manager` in the `open-cluster-management-observability` namespace. This operator will deploy the new metrics collectors based on the PrometheusAgent to the managed clusters.

thibaultmg commented on observability/obs_enable_mcoa.adoc:
```diff
@@ -0,0 +1,54 @@
++
+[source,bash]
+----
+oc patch MultiClusterObservability observability -n open-cluster-management-observability --type=merge -p '{"spec":{"platform":{"enabled": true},"userWorkloads":{"enabled": true}}}'
```
oc patch mco observability -n open-cluster-management-observability --type=merge -p '{"spec":{"capabilities":{"platform":{"metrics":{"default":{"enabled": true}}},"userWorkloads":{"metrics":{"default":{"enabled": true}}}}}}'

thibaultmg commented on observability/obs_enable_mcoa.adoc:
```diff
@@ -0,0 +1,54 @@
++
+[source,bash]
+----
+oc get clustermanagementaddon -n multicluster-observability-addon
```
What about:
oc get cma multicluster-observability-addon -o yaml | yq '.spec.installStrategy.placements'

thibaultmg commented on observability/obs_mcoa_add_custom_metrics.adoc:
```diff
@@ -0,0 +1,42 @@
+
+. Create a new `ScrapeConfig` resource in the `open-cluster-management-observability` namespace that includes values for the required parameters, `jobName`, `metricsPath`, and `params`.
+
+. Add the appropriate label for the `app.kubernetes.io/component` parameter to specify whether the metrics are for platform monitoring or user workload monitoring. Use one of the following label values, `platform-metrics-collector` or `user-workload-metrics-collector`.
```
Maybe add the when `platform-metrics-collector` is set, MCOA will automatically set the `scrapeClass` and the `targets` to enable federation from the in-cluster platform prometheus. While when `user-workload-metrics-collector`, they will be automatically set to target the in-cluster user-workload monitoring. But this can be overridden by the user in the latter case.

dockerymick replied:
Is there an example that we can share with the user? Maybe a YAML snippet ? Also what does "in-cluster" mean? Are you referring to the hub cluster?

thibaultmg replied:
"in-cluster platform prometheus" refers to the ocp monitoring prometheus, that operates for this single cluster.

Here is an example:

```yaml
apiVersion: monitoring.rhobs/v1alpha1
kind: ScrapeConfig
metadata:
  name: platform-metrics-alerts
  namespace: open-cluster-management-agent-addon
spec:
  jobName: alerts
  metricRelabelings:
    - action: labeldrop
      regex: managed_cluster|id
  metricsPath: /federate
  params:
    'match[]':
      - '{__name__="ALERTS"}'
  scheme: HTTPS
  scrapeClass: ocp-monitoring
  staticConfigs:
    - targets:
        - 'prometheus-k8s.openshift-monitoring.svc:9091'
```

jobName, params and metricsPath is set by the user while the other fields are enforced by the operator. But for metricRelabelings, the user can still add his own relabellings on top of ours.

thibaultmg replied:
And for UWL scrapeconfigs, we don't enforce scrapeClass and  staticConfigs. We only set a default that targets the ocp uwl monitoring prometheus. But the user can override that.

thibaultmg commented on observability/obs_mcoa_add_custom_metrics.adoc:
```diff
@@ -0,0 +1,42 @@
+. Add the `ScrapeConfig` resource reference to the placements of the `ClusterManagementAddOn` resource where you want the resource to be deployed.
+
++
+*Note:* Ensure that you reference the `ScrapeConfig` resource only after each federated metric is deployed on a given managed cluster.
```
I suppose you mean:
Ensure that you reference the `ScrapeConfig` resource only after it has been created. Otherwise, the addon status will become `Deploying` as the resource is not found.

thibaultmg commented on observability/obs_mcoa_add_custom_metrics.adoc:
```diff
@@ -0,0 +1,42 @@
+  name: add-custom-metrics
+  namespace: open-cluster-management-observability
+  labels:
+    - app.kubernetes.io/component: <platform-metrics-collector>
```
Let's remove the <>, no?

thibaultmg commented on observability/obs_mcoa_config_apis.adoc:
```diff
@@ -0,0 +1,102 @@
+[#config-apis-mcoa]
+= Configuring APIs for the multicluster obsservability add-on
+
+Configure the default metrics specific APIs of the multicluster observability add-on. When you enable the multicluster observability add-on, placements are automatically created for your managed clusters.
```
We don't create placements. We only use the global placement that already exist by default. But we autimatically add the default configuration resources to this placement in the ClusterManagementAddOn.

thibaultmg commented on observability/obs_mcoa_config_apis.adoc:
```diff
@@ -0,0 +1,102 @@
+
+Configure the default metrics specific APIs of the multicluster observability add-on. When you enable the multicluster observability add-on, placements are automatically created for your managed clusters.
+
+When you reference a new placement in your `ClusterManagementAddon` resource, the `multicluster-observability-addon-manager` creates a specific default `PrometheusAgent` resource and adds it to the referenced placement. 
```
We also add the required ScrapeConfigs and PrometheusRules.

thibaultmg commented on observability/obs_mcoa_config_apis.adoc:
```diff
@@ -0,0 +1,102 @@
+
+Complete the following to configure the APIs for the multicluster observability add-on:
+
+. Configure the `PrometheusAgent` resource by defining the prometheus server for federating metrics from the default scrape interval. Complete the following steps:
```
Let's discuss this part tomorrow. There are things to change.

dockerymick committed:
Revise MultiClusterObservability documentation and commands

Updates after dev review

dockerymick committed:
Updates and hidden comments after dev review

Corrected label value and added notes for clarity.

dockerymick committed:
Updates after dev review

dockerymick committed:
remove hidden comment

dockerymick committed:
Remove anchors from H2 sections

dockerymick committed:
Live review with Thibault

Refine descriptions and notes for clarity and consistency.

dockerymick committed:
More from the live review

Corrected spelling of 'Exportng' to 'Exporting' and changed '2 hours' to 'two hours' for consistency.

dockerymick committed:
More updates during live review

dockerymick committed:
More updates during live review

Updated text for clarity and corrected terminology in the documentation regarding the multicluster observability add-on.

dockerymick commented on observability/obs_mcoa_config_apis.adoc:
```diff
@@ -0,0 +1,66 @@
+
+. *Optional* Override the default scrape interval for your `PrometheusAgent` resource by changing the `scrapeInterval` parameter. The default value is `300s`. You can also override the `scrapeInterval` of the `scrapeConfig` resource.
+
+. Configure the `ScrapeConfig` resource to define a set of metrics for independent federation from Prometheus of the hub cluster. Complete the following steps:
```
@thibaultmg mentioning here bc this is where we stopped before the meeting ended

thibaultmg replied:
```suggestion
. Configure the `ScrapeConfig` resource to define a set of metrics for independent federation from Prometheus of your managed clusters. Complete the following steps:
```

thibaultmg commented on observability/export_metrics_ext_endpts.adoc:
```diff
@@ -0,0 +1,44 @@
+[#export-metrics-external]
+= Exporting metrics to external endpoints for the multicluster observability add-on
+
+Configure your custom `remoteWrite` configurations to your `PrometheusAgents` resources. Metrics are directly sent by the managed cluster to the external endpoint when you configure your `remoteWrite` configuration. Filter metrics by using relabeling rules. Enhance resiliency during network partitions between your managed and hub clusters for up to two hours.
```
We should precise that this is an alternative to https://redhat-internal.slack.com/archives/C03A998ETHR/p1763655471769439
Where instead of having the hub forwarding the metrics, this is done directly from the managed clusters, with more configuration possibilities and better resiliency to network partitions.

dockerymick replied:
Why is it important to say that this is an alternative to forwarding to the hub cluster? It is best to present it as is (a task that the user can choose to complete if that is their preference). Is hub cluster forwarding the default behavior? From my perspective, we are highlighting the benefits of exporting metrics. Let me know if it is still unclear.

thibaultmg commented on observability/obs_enable_mcoa.adoc:
```diff
@@ -0,0 +1,52 @@
+
+Enable the multicluster observability add-on on your hub cluster to configure your metrics for your platform and user workloads. You are required to enable platform workloads and it is optional to enable user workloads.
+
+When you enable `platform` and `userWorkloads` specifications, the `MultiClusterObservability` operator stops deploying the metrics collectors to your managed clusters and deploys the `multicluster-observability-addon-manager` in the `open-cluster-management-observability` namespace. The `MultiClusterObservability` operator deploys the new metrics collectors based on the `PrometheusAgent` resourc ...
```
```suggestion
When you enable `platform` and `userWorkloads` specifications, the `MultiClusterObservability` operator stops deploying the metrics collectors to your managed clusters and deploys the `multicluster-observability-addon-manager` in the `open-cluster-management-observability` namespace. The `multicluster-observability-addon-manager` deploys the new metrics collectors based on the `PrometheusAgent` resource defined on your hub cluster.
```

thibaultmg commented on observability/obs_enable_mcoa.adoc:
```diff
@@ -0,0 +1,52 @@
+. To verify that the default configuration resources for the multicluster observability add-on are created, open your `multicluster-observability-addon` `ClusterManagementAddon` resource. Run the following command:
+
++
+[source,bash]
```
I would replace this command with:
oc get prometheusagents -n open-cluster-management-observability

And then we can add: 

You can also verify that the default configurations are added to your placements with:
oc get cma multicluster-observability-addon -o yaml | yq '.spec.installStrategy.placements'

thibaultmg commented on observability/obs_mcoa_add_custom_metrics.adoc:
```diff
@@ -0,0 +1,45 @@
+[#add-custom-metrics-mcoa]
+= Adding custom metrics for the multicluster observability add-on
+
+Add your own custom metrics to be collected from your managed clusters by configuring the `ScrapeConfig` resource. The new `ScrapeConfig` resource is deployed to the managed clusters that you specify in the placements of the `ClusterManagementAddOn` resource. The Prometheus operator on each managed cluster adds the new `ScrapeConfig` resource to the `PrometheusAgent` resource.
```
```suggestion
Add your own custom metrics to be collected from your managed clusters by configuring the `ScrapeConfig` resource. The new `ScrapeConfig` must be added to the `Placement` configurations of the `ClusterManagementAddOn` resource for deploying on the corresponding managed clusters. The Prometheus operator on each managed cluster adds the new `ScrapeConfig` resource to the `PrometheusAgent` resource.
```

thibaultmg commented on observability/obs_mcoa_config_apis.adoc:
```diff
@@ -0,0 +1,66 @@
+[#config-apis-mcoa]
+= Configuring APIs for the multicluster observability add-on
+
+Configure the default metrics-specific APIs of the multicluster observability add-on. When you reference a new placement in your `ClusterManagementAddon` resource, the `multicluster-observability-addon-manager` automatically creates specific default configuration resources and creates the configuration resources to the referenced placement. The following resources are the default configuration re ...
```
```suggestion
Configure the default metrics-specific APIs of the multicluster observability add-on. When you reference a new placement in your `ClusterManagementAddon` resource, the `multicluster-observability-addon-manager` automatically creates specific default `PrometheusAgent` resources. The `multicluster-observability-addon-manager` adds the `PrometheusAgent` resources reference in the related `Placement` configurations.

While there is one `PrometheusAgent` resource created by the placement, the default `ScrapeConfigs` and `PrometheusRules` are common to all placements. The following resources are the default configuration resources for the multicluster observability add-on: `PrometheusAgent, ScrapeConfigs`, and `PrometheusRules`.
```

thibaultmg commented on observability/obs_mcoa_config_apis.adoc:
```diff
@@ -0,0 +1,66 @@
+    - '{__name__="up"}'
+----
+
+.. Define the target prometheus source server for platform monitring and user workloads. Your `ScrapeConfig` resource might resemble the following YAML file example:
```
The user can only customized this target for UWL monitoring. We set a default pointing to the OCP uwl prometheus.
For platform monitoring, the user must not update these fields, they are enforced by MCOA.

thibaultmg commented on observability/obs_mcoa_config_apis.adoc:
```diff
@@ -0,0 +1,66 @@
+.. Define the target prometheus source server for platform monitring and user workloads. Your `ScrapeConfig` resource might resemble the following YAML file example:
+
++
+[source,yaml]
```
I would remove this.

thibaultmg commented on observability/obs_mcoa_config_apis.adoc:
```diff
@@ -0,0 +1,66 @@
+. Configure the `PrometheusRule` resource to define alerting and recording rules for custom workloads on managed clusters.
+
++
+*Note:* Be sure to use the `monitoring.coreos.com` group for the `PrometheusRule` resource.
```
I think we should have a specific document for configuring rules. Or we should at least get this content here: https://docs.google.com/document/d/1joCg1kN4yd6HZ_GzJTRYXjNPvJGz_p4jhw5r26zYCI4/edit?tab=t.0#heading=h.5abtripm1var
There are specifics for uwl to be described. And limitations.

dockerymick replied:
If there are limitations, it should be added to the known issues for Observability. A new issue must be created to add new known issues.

dockerymick replied:
What parts of your draft need to be mentioned?

dockerymick replied:
Made a couple updates

thibaultmg commented on observability/obs_mcoa_config_apis.adoc:
```diff
@@ -0,0 +1,66 @@
+
+. Configure the `AddonDeploymentConfig` resource to customize the deployment of the multicluster observability add-on on managed clusters. *Note:* The values in the `AddonDeploymentConfig` resource are override direct modifications of other resources. Complete the following steps:
+
+.. Define the namespace where you install the `PrometheusAgent` resource for the `agentInstallNamespace` parameter.
```
We can add that by default it is `open-cluster-management-agent-addon`

thibaultmg commented on observability/obs_mcoa_federate.adoc:
```diff
@@ -0,0 +1,32 @@
+[#federate-uw-coo]
+= Federating user workloads from Cluster Observability Operator
+
+Federate your user workloads from the Red Hat OpenShift Cluster Observability Operator on your managed clusters to scale your metrics. By default, user workload metrics are federated from the `MultiClusterObservability` resource.
```
```suggestion
Federate your user workloads from the Red Hat OpenShift Cluster Observability Operator on your managed clusters. By default, user workload metrics are federated from the Prometheus user-workload on your {ocp-short} cluster.
```

thibaultmg commented on observability/obs_mcoa_intro.adoc:
```diff
@@ -0,0 +1,68 @@
+
+Control monitoring behavior on your managed clusters by configuring default resource configurations. The following default resources are automatically created for platform monitoring and to have functional dashboards, `PrometheusAgent`, `ScrapeConfig`, and `PrometheusRule`. View the following descriptions of each API:
+
+`PrometheusAgent`:: Defines an agent-focused Prometheus deployment optimized for remote-write workflows. Configures remote-write targets, scrape intervals, and resource limits. Use the `PrometheusAgent` resource as your metrics collector.
```
```suggestion
`PrometheusAgent`:: Defines a Prometheus Agent deployment, a lightweight Prometheus. Configures remote-write targets, scrape intervals, and resource limits. Use the `PrometheusAgent` resource as your metrics collector.
```

thibaultmg commented on observability/obs_mcoa_intro.adoc:
```diff
@@ -0,0 +1,68 @@
+
+`PrometheusAgent`:: Defines an agent-focused Prometheus deployment optimized for remote-write workflows. Configures remote-write targets, scrape intervals, and resource limits. Use the `PrometheusAgent` resource as your metrics collector.
+
+`ScrapeConfig`:: A list of metrics to be federated from a given `target`. Configure `job_name`, `metrics_path`, `params`, `relabel_configs`, and TLS settings. You can override the `target` configuration for user workloads.
```
```suggestion
`ScrapeConfig`:: A list of metrics to be federated from a given `target`. Configure `job_name`, `metrics_path`, `params` and `relabel_configs`. You can override the `target` configuration for user workloads.
```

thibaultmg commented on observability/obs_mcoa_intro.adoc:
```diff
@@ -0,0 +1,68 @@
+
+`PrometheusRule`:: Supports alerting and recording rule definitions that you can apply for each cluster. Supports groups of rules with evaluation intervals and alert annotations or labels.
+
+`ClusterManagementAddOn`:: Resources include an `AddonDeploymentConfig` to customize the deployment of the multicluster observability add-on on managed clusters. You can use this configuration the node placement and select the target namespace of the managed cluster, where the Prometheus Agent pods are deployed.
```
```suggestion
`ClusterManagementAddOn`:: Placements and configurations to be deployed to the managed clusters. Configurations include an `AddonDeploymentConfig` to customize the deployment of the multicluster observability add-on on managed clusters. You can use this configuration to customize the node placement and select the intstallation namespace on the managed clusters, where the Prometheus Agent pods are deployed.
```

thibaultmg commented on observability/obs_mcoa_intro.adoc:
```diff
@@ -0,0 +1,68 @@
+
+== Default platform metrics
+
+By default, the multicluster observability add-on collects platform metrics from your managed clusters. There are a set of `ScrapeConfig` resources that are automatically generated and deployed to all of your managed clusters. All placements that you reference within the `ClusterManagementAddOn` resource automatically receive the `ScrapeConfig` configurations.
```
I would add that these default scrapeConfigs enable fully operational platform dashboards.

thibaultmg commented on observability/obs_mcoa_intro.adoc:
```diff
@@ -0,0 +1,68 @@
+The following alerting rules are pushed to all managed clusters:
+
+`MetricsCollectorNotIngestingSamples`:: Alerts you when the `PrometheusAgent` resource is not federating any metrics.
+`MetricsCollectorRemoteWriteFailures`:: Alerts you when there is a high failure rate for remote-write requests in the `PrometheusAgent` resource of your hub cluster.
```
```suggestion
`MetricsCollectorRemoteWriteFailures`:: Alerts you when there is a high failure rate for remote-write requests in the `PrometheusAgent` resource of the managed cluster.
```

thibaultmg commented on observability/obs_mcoa_relabel.adoc:
```diff
@@ -0,0 +1,27 @@
+[#relabeling-mcoa-default-metrics]
+= Relabeling default metrics for the multicluster observability add-on
+
+Relabel configurations to modify the metrics that you are collecting. Relabel configurations in each `ScrapeConfig` resource or globally in the `remoteWrite` configuration of the `PrometheusAgent` resource.
```
```suggestion
Relabel configurations to modify the default metrics that you are collecting. Relabel configurations in each `ScrapeConfig` resource or globally in the `remoteWrite` configuration of the `PrometheusAgent` resource. This can be used to remove some metrics that are collected by default for plaftform monitoring.
```

thibaultmg commented on observability/obs_mcoa_relabel.adoc:
```diff
@@ -0,0 +1,27 @@
+
+- You have installed and enabled the multicluster observability add-on. For more information, see xref:../observability/obs_enable_mcoa.adoc#enable-mcoa[Enabling the multicluster observability add-on].
+
+Complete the following steps to relabel default metrics for the multicluster observability add-on:
```
```suggestion
Complete the following steps to remove a metric collected by default, using relabelling:
```

thibaultmg commented on observability/obs_mcoa_relabel.adoc:
```diff
@@ -0,0 +1,27 @@
+
+Complete the following steps to relabel default metrics for the multicluster observability add-on:
+
+. Remove the `Watchdog` alert metric by adding the `metricRelabelings` in the `platform-metrics-alerts` `ScrapeConfig` resource where the `Watchdog` metric is defined.
```
```suggestion
. Remove the `Watchdog` alert metric by adding the `metricRelabelings` in the `platform-metrics-alerts` `ScrapeConfig` resource where the `ALERT` metric is defined.
```

thibaultmg commented on observability/obs_mcoa_relabel.adoc:
```diff
@@ -0,0 +1,27 @@
+
+. Remove the `Watchdog` alert metric by adding the `metricRelabelings` in the `platform-metrics-alerts` `ScrapeConfig` resource where the `Watchdog` metric is defined.
+//is there an exmaple of this? ^
+. To remove the `Watchdog` alert metric, add the `writeRelabelConfigs` specification to the `remoteWrite` configuration of the `PrometheusAgent` resource. The metric removal is complete after it is federated. Your `PrometheusAgent` resource might resemble the following YAML file:
```
I would use the ScrapeConfig rather. Here is an example:

```yaml
apiVersion: monitoring.rhobs/v1alpha1
kind: ScrapeConfig
metadata:
  name: platform-metrics-alerts
  namespace: open-cluster-management-observability
spec:
  jobName: alerts
  metricRelabelings:
    - action: labeldrop
      regex: managed_cluster|id
    - action: drop
      regex: ^Watchdog$
      sourceLabels:
        - alertname
  metricsPath: /federate
  params:
    'match[]':
      - '{__name__="ALERTS"}'
  scheme: HTTPS
  scrapeClass: not-configurable
  staticConfigs:
    - targets:
        - not-configurable
```

dockerymick committed:
Apply suggestion from @thibaultmg

Co-authored-by: Thibault Mange <22740367+thibaultmg@users.noreply.github.com>

dockerymick committed:
Update commands after dev review

Added command to check Prometheus agents in the documentation.

dockerymick committed:
Apply suggestion from @thibaultmg

Co-authored-by: Thibault Mange <22740367+thibaultmg@users.noreply.github.com>

dockerymick committed:
Apply suggestion from @thibaultmg

Co-authored-by: Thibault Mange <22740367+thibaultmg@users.noreply.github.com>

dockerymick committed:
Apply suggestion from @thibaultmg

Co-authored-by: Thibault Mange <22740367+thibaultmg@users.noreply.github.com>

dockerymick commented on observability/obs_mcoa_config_apis.adoc:
```diff
@@ -0,0 +1,68 @@
+  - targets:
+    - prometheus-k8s.openshift-monitoring.svc:9091
+    - prometheus-user-workload.openshift-user-workload-monitoring.svc:9092
+----
```
```suggestion
```

dockerymick committed:
Apply suggestion from @thibaultmg

Co-authored-by: Thibault Mange <22740367+thibaultmg@users.noreply.github.com>

dockerymick committed:
Update ScrapeConfig for platform metrics alerts

dockerymick committed:
Revise export metrics documentation for clarity

Updated instructions for configuring external metrics endpoints to improve clarity and detail.

dockerymick committed:
Updates after dev review

Clarified note about the 'platform-metrics-collector' label and added information on overriding parameters.

dockerymick committed:
Update obs_mcoa_config_apis.adoc

dockerymick committed:
changes after dev review

dockerymick committed:
update after dev review

dockerymick committed:
Apply suggestion from @thibaultmg

Co-authored-by: Thibault Mange <22740367+thibaultmg@users.noreply.github.com>

dockerymick commented on observability/obs_mcoa_config_apis.adoc:
```diff
@@ -0,0 +1,57 @@
+
+. Configure the `AddonDeploymentConfig` resource to customize the deployment of the multicluster observability add-on on managed clusters. *Note:* The values in the `AddonDeploymentConfig` resource are override direct modifications of other resources. Complete the following steps:
+
+.. Define the namespace where you install the `PrometheusAgent` resource for the `agentInstallNamespace` parameter.
```
```suggestion
.. Define the namespace where you install the `PrometheusAgent` resource for the `agentInstallNamespace` parameter. The default namespace is `open-cluster-management-agent-addon`.
```

dockerymick committed:
Apply suggestion from @dockerymick

jacobbaungard commented on observability/obs_mcoa_intro.adoc:
```diff
@@ -0,0 +1,68 @@
+[#obs-mcoa-intro]
+= Multicluster observability add-on
+
+Use the multicluster observability add-on to configure metrics collection by using names, matchers, and recording rules for both platform and user workloads. Edit your `ClusterManagementAddOn` resource, scale metric federation, and reduce data loss during network partitions between your managed and hub clusters.
```
Is this the very first sentence in this section? Shouldn't we explain a bit what the changes between enabling this and the default mode is?

To me it almost sounds like we're not collecting metrics at all, if we don't enable this.

dockerymick replied:
If that will help the user then sure. Can you help me understand the "default" mode

jacobbaungard replied:
The "default" mode is what is already released and what we already document in the rest of the doc.

In 2.15 we release this alternative method of metric collection. The differences are well explained in Thibaults original document.

dockerymick replied:
OK, so you are requesting that I specifically write that this is an alternative way to collect metrics? From my perspective, the user assumes that metrics can be collected because they must install Observability anyways.

jacobbaungard replied:
> From my perspective, the user assumes that metrics can be collected because they must install Observability anyways.

I don't follow. If you install Observability using the normal installation instruction metrics _will_ be collected.

If they want to use the _new_ method of metric collection, which replaces the "old" way of collecting metrics, they need to specifically enable that (which is explained in `Enabling the multicluster observability add-on`).

It should be clear in the introduction here that:

1) MCOA is a new way of collecting metrics from managed clusters
2) That it is not enabled by default
3) The main advantages/differences

This section is super confusing as it stands here.

dockerymick replied:
Thanks for providing your perspective and suggestions. I made updates

oafischer commented on observability/export_metrics_ext_endpts.adoc:
```diff
@@ -0,0 +1,44 @@
+[#export-metrics-external]
+= Exporting metrics to external endpoints for the multicluster observability add-on
+
+To configure an external metrics endpoint, add your custom `remoteWrite` specification to your `PrometheusAgents` resources. When you configure your `remoteWrite` specification, metrics are directly sent by the managed cluster to the external endpoint. Update related multicluster observability add-on configurations by relabeling default metrics. Exporting metrics from your managed cluster helps y ...
```
```suggestion
To configure an external metrics endpoint, add your custom `remoteWrite` specification to your `PrometheusAgents` resources. When you configure your `remoteWrite` specification, the managed cluster sends metrics directly to the external endpoint.

Update related multicluster observability add-on configurations by relabeling default metrics. Exporting metrics from your managed cluster helps you improve resiliency during network partitions between your managed and hub clusters for up to two hours.
```

Suggestion for active voice

I also think two paragraphs might help with readability here.

oafischer commented on observability/export_metrics_ext_endpts.adoc:
```diff
@@ -0,0 +1,44 @@
+.Prerequisites
+
+- You have installed and enabled the multicluster observability add-on. For more information, see xref:../observability/obs_enable_mcoa.adoc#enable-mcoa[Enabling the multicluster observability add-on].
+
```
```suggestion

.Procedure

```

oafischer commented on observability/obs_enable_mcoa.adoc:
```diff
@@ -0,0 +1,60 @@
+
+- You must enable the Observability service on your hub cluster. For more details, see xref:../observability/observability_enable.adoc[Enabling the Observability service].
+- You must install the Red Hat OpenShift Cluster Observability Operator. For more information, see link:https://docs.redhat.com/en/documentation/red_hat_openshift_cluster_observability_operator/1-latest/html-single/about_red_hat_openshift_cluster_observability_operator/index[Cluster Observability Operator overview].
+
```
```suggestion

.Procedure

```

oafischer commented on observability/obs_mcoa_add_custom_metrics.adoc:
```diff
@@ -0,0 +1,47 @@
+[#add-custom-metrics-mcoa]
+= Adding custom metrics for the multicluster observability add-on
+
+Add your own custom metrics to be collected from your managed clusters by configuring the `ScrapeConfig` resource. The new `ScrapeConfig` must be added to the `Placement` configurations of the `ClusterManagementAddOn` resource for deploying on the corresponding managed clusters. The Prometheus operator on each managed cluster adds the new `ScrapeConfig` resource to the `PrometheusAgent` resource.
```
Just thinking ahead, if we say something like "new", it will be outdated by the next version, unless we remember to come back to this and change it. Should we just leave it out now? Let me know what you think.

dockerymick replied:
Thats a great point. I'll remove new

oafischer commented on observability/obs_mcoa_add_custom_metrics.adoc:
```diff
@@ -0,0 +1,47 @@
+. Add the appropriate label for the `app.kubernetes.io/component` parameter to specify whether the metrics are for platform monitoring or user workload monitoring. Use one of the following label values, `platform-metrics-collector` or `user-workload-metrics-collector`.
+
++
+*Note:* When you use the `platform-metrics-collector` label, the multicluster observability add-on automatically set the `scrapeClass` and `targets` parameters to enable federation from the platform Prometheus of your {ocp-short} managed cluster. You can override the `scrapeClass` and `targets` parameters by adding the value that you need.
```
```suggestion
*Note:* When you use the `platform-metrics-collector` label, the multicluster observability add-on automatically sets the `scrapeClass` and `targets` parameters to enable federation from the platform Prometheus of your {ocp-short} managed cluster. You can override the `scrapeClass` and `targets` parameters by adding the value that you need.
```

oafischer commented on observability/obs_mcoa_config_apis.adoc:
```diff
@@ -0,0 +1,57 @@
+
+Configure the default metrics-specific APIs of the multicluster observability add-on. When you reference a new placement in your `ClusterManagementAddon` resource, the `multicluster-observability-addon-manager` automatically creates specific default `PrometheusAgent` resources. The `multicluster-observability-addon-manager` adds the `PrometheusAgent` resources reference in the related `Placement` ...
+
+While there is one `PrometheusAgent` resource created by the placement, the default `ScrapeConfigs` and `PrometheusRules` are common to all placements. The following resources are the default configuration resources for the multicluster observability add-on: `PrometheusAgent, ScrapeConfigs`, and `PrometheusRules`.
```
```suggestion
While there is one `PrometheusAgent` resource created by the placement, the default `ScrapeConfigs` and `PrometheusRules` are common to all placements. The following resources are the default configuration resources for the multicluster observability add-on: `PrometheusAgent`, `ScrapeConfigs`, and `PrometheusRules`.
```

oafischer commented on observability/obs_mcoa_config_apis.adoc:
```diff
@@ -0,0 +1,57 @@
+.Prerequisites
+
+- You have installed and enabled the multicluster observability add-on. For more information, see xref:../observability/obs_enable_mcoa.adoc#enable-mcoa[Enabling the multicluster observability add-on].
+
```
```suggestion

.Procedure

```

oafischer commented on observability/obs_mcoa_federate.adoc:
```diff
@@ -0,0 +1,32 @@
+- User workload monitoring is enabled in the    `MultiClusterObservability` custom resource.
+- You have created `ScrapeConfig` resources with the `app.kubernetes.io/component: <user-workload-metrics-collector>` label.
+- The `ScrapeConfig` resources are referenced in the configurations list of the `ClusterManagementAddOn` for the target placements.
+
```
```suggestion

.Procedure

```

oafischer commented on observability/obs_mcoa_intro.adoc:
```diff
@@ -0,0 +1,68 @@
+
+== Managed cluster workloads
+
+Managed cluster monitoring workloads use the Prometheus Operator and Prometheus Agent from the cluster observability operator. The multicluster observability add-on utilizes the existing `obo-prometheus-operator` from Red Hat OpenShift Cluster Observabililty Operator on a managed cluster to reconcile Prometheus custom resources. If `obo-prometheus-operator` is not present, the multicluster observ ...
```
```suggestion
Managed cluster monitoring workloads use the Prometheus Operator and Prometheus Agent from the cluster observability operator. The multicluster observability add-on uses the existing `obo-prometheus-operator` from {ocp} Cluster Observabililty Operator on a managed cluster to reconcile Prometheus custom resources. If `obo-prometheus-operator` is not present, the multicluster observability add-on deploys `obo-prometheus-operator` within the configured `agentInstallNamespace`.
```

dockerymick replied:
So the "Container Platform" part is not used in the product doc: https://docs.redhat.com/en/documentation/red_hat_openshift_cluster_observability_operator/1-latest/html-single/about_red_hat_openshift_cluster_observability_operator/index

That's why I didn't use the ocp conref. However, we should add a conref for Red Hat OpenShift Cluster Observability Operator. Accepting the change: utilize ---> uses

oafischer commented on observability/obs_mcoa_intro.adoc:
```diff
@@ -0,0 +1,68 @@
+
+Both of the multicluster observability add-on and Cluster Observabililty Operator share the `monitoring.rhobs` API Group for the `ScrapeConfig` and `PrometheusAgent` resources.
+
+For other managed clusters that are not OpenShift clusters, the following resources are deployed:
```
```suggestion
For other managed clusters that are not {ocp-short} clusters, the following resources are deployed:
```

dockerymick replied:
great catch!

oafischer commented on observability/obs_mcoa_relabel.adoc:
```diff
@@ -0,0 +1,42 @@
+.Prerequisites
+
+- You have installed and enabled the multicluster observability add-on. For more information, see xref:../observability/obs_enable_mcoa.adoc#enable-mcoa[Enabling the multicluster observability add-on].
+
```
```suggestion

.Procedure

```

dockerymick committed:
updates after peer review

dockerymick committed:
Update obs_enable_mcoa.adoc

dockerymick committed:
Update obs_mcoa_add_custom_metrics.adoc

thibaultmg commented on observability/obs_mcoa_config_apis.adoc:
```diff
@@ -0,0 +1,57 @@
+
+. Configure the `PrometheusRule` resource to limit the cardinality of your collected metrics on your hub cluster. Complete the following steps:
+
+.. Define alerting and recording rules for custom workloads on your managed clusters.
```
They can be defined for both platform and uwl monitoring. 

As stated in my document:

------

For Prometheus rules targeting user-workloads, you must include the specific annotation observability.open-cluster-management.io/target-namespace to designate the namespace where the rule should be deployed.

Failure to include this annotation will result in the rules being deployed to the default installation namespace. This default namespace contains the openshift.io/cluster-monitoring=true label, which means the rules will not be monitored by the in-cluster user-workload monitoring stack.

-----

dockerymick replied:
Added within the notes. Let me know what you think

thibaultmg commented on observability/obs_mcoa_config_apis.adoc:
```diff
@@ -0,0 +1,57 @@
+. Configure the `PrometheusRule` resource to limit the cardinality of your collected metrics on your hub cluster. Complete the following steps:
+
+.. Define alerting and recording rules for custom workloads on your managed clusters.
+.. To define the namespace where you want to deploy your `PrometheusRule` resource, add the following annotation: `observability.open-cluster-management.io/target-namespace`.
```
We should explain why I think, as described in my above comment.

dockerymick committed:
Update observability/obs_mcoa_add_custom_metrics.adoc

Co-authored-by: Oliver Fischer <108426932+oafischer@users.noreply.github.com>

dockerymick committed:
Update observability/obs_mcoa_config_apis.adoc

Co-authored-by: Oliver Fischer <108426932+oafischer@users.noreply.github.com>

dockerymick commented on observability/obs_mcoa_intro.adoc:
```diff
@@ -0,0 +1,68 @@
+
+The multicluster observability add-on health status is reported through the `ManagedClusterAddOn` resource. From the {ocp-short} console for each managed cluster, the `ManagedClusterResource` is displayed in the *Fleet Management* view. The status is degraded if some resources are not deployed or the platform `PrometheusAgent` resource is not running.
+
+*Note:* For {ocp-short} versions earlier than version 4.20, select *All Clusters* from the cluster switcher.
```
```suggestion
*Note:* For {ocp-short} versions earlier than version 4.20, select *All Clusters* from the *Cluster* switcher.
```

dockerymick committed:
Apply suggestion from @dockerymick

dockerymick committed:
Merge branch '2.15_stage' into mj-ACM-23917

dockerymick committed:
Update obs_mcoa_config_apis.adoc

dockerymick committed:
Update obs_mcoa_federate.adoc

dockerymick commented on observability/obs_mcoa_intro.adoc:
```diff
@@ -0,0 +1,68 @@
+
+== Managed cluster workloads
+
+Managed cluster monitoring workloads use the Prometheus Operator and Prometheus Agent from the cluster observability operator. The multicluster observability add-on utilizes the existing `obo-prometheus-operator` from Red Hat OpenShift Cluster Observabililty Operator on a managed cluster to reconcile Prometheus custom resources. If `obo-prometheus-operator` is not present, the multicluster observ ...
```
```suggestion
Managed cluster monitoring workloads use the Prometheus Operator and Prometheus Agent from the cluster observability operator. The multicluster observability add-on uses the existing `obo-prometheus-operator` from Red Hat OpenShift Cluster Observabililty Operator on a managed cluster to reconcile Prometheus custom resources. If `obo-prometheus-operator` is not present, the multicluster observability add-on deploys `obo-prometheus-operator` within the configured `agentInstallNamespace`.
```

dockerymick committed:
Apply suggestion from @dockerymick

dockerymick committed:
Update obs_mcoa_intro.adoc

dockerymick committed:
Update obs_mcoa_relabel.adoc

dockerymick committed:
Update obs_mcoa_config_apis.adoc

jacobbaungard commented on release_notes/acm_new.adoc:
```diff
@@ -109,6 +109,8 @@ Learn about new features for application management.
 
 - Use `RightSizingRecommendation` to optimize workloads and understand resource usage across your virtual machines. To learn more, see link:../observability/obs_right_size_intro.adoc#right-size-intro[`RightSizingRecommendation` guides (Technology Preview)].
 
+- Enable the multicluster observability add-on to collect platform metrics and user workloads from your managed clusters. For more information, see link:../observability/obs_mcoa_intro.adoc#obs-mcoa-intro[Multicluster observability add-on].
```
It sounds like it was not possible to collect these metrics before. It definitely was.

MCOA introduces a new method of metrics collection, but we collect (more or less) the same metrics as before. Just using different set of technologies.

jacobbaungard commented on observability/obs_mcoa_intro.adoc:
```diff
@@ -0,0 +1,68 @@
+
+== Managed cluster workloads
+
+Managed cluster monitoring workloads use the Prometheus Operator and Prometheus Agent from the cluster observability operator. The multicluster observability add-on uses the existing `obo-prometheus-operator` from Red Hat OpenShift Cluster Observabililty Operator on a managed cluster to reconcile Prometheus custom resources. If `obo-prometheus-operator` is not present, the multicluster observabil ...
```
```suggestion
Managed cluster monitoring workloads use the Prometheus Operator and Prometheus Agent from the Cluster Observability Operator. The multicluster observability add-on uses the existing `obo-prometheus-operator` from Red Hat OpenShift Cluster Observabililty Operator on a managed cluster to reconcile Prometheus custom resources. If `obo-prometheus-operator` is not present, the multicluster observability add-on deploys `obo-prometheus-operator` within the configured `agentInstallNamespace`.
```

dockerymick committed:
Update observability/obs_mcoa_intro.adoc

Co-authored-by: Jacob Baungård Hansen <jacobbaungard@redhat.com>

dockerymick committed:
Improving intro description

dockerymick committed:
IMproving what's new entry

thibaultmg commented on observability/obs_mcoa_config_apis.adoc:
```diff
@@ -0,0 +1,59 @@
+
++
+*Notes:* 
+- When the `observability.open-cluster-management.io/target-namespace` is not set in your `PrometheusRule` resource, the `PrometheusRule` resource are deployed to the default installation namespace. If `openshift.io/cluster-monitoring` is set to `true`, the `PrometheusRule` resouces are monitored by {ocp-short} user workload monitoring stack.
```
```suggestion
- When the `observability.open-cluster-management.io/target-namespace` is not set in your `PrometheusRule` resource, the `PrometheusRule` resource are deployed to the default installation namespace. If `openshift.io/cluster-monitoring` is set to `true`, the `PrometheusRule` resouces are not monitored by {ocp-short} user workload monitoring stack.
```

dockerymick committed:
Update observability/obs_mcoa_config_apis.adoc

Co-authored-by: Thibault Mange <22740367+thibaultmg@users.noreply.github.com>

thibaultmg reviewed (approved):
/lgtm

jacobbaungard reviewed (approved):
/lgtm

dockerymick commented:
Thanks everyone! This was a large effort! I appreciate yall!

swopebe commented on observability/export_metrics_ext_endpts.adoc:
```diff
@@ -0,0 +1,48 @@
+[#export-metrics-external]
+= Exporting metrics to external endpoints for the multicluster observability add-on
```
We need a short description:

Why do this?

swopebe commented on observability/export_metrics_ext_endpts.adoc:
```diff
@@ -0,0 +1,48 @@
+[#export-metrics-external]
+= Exporting metrics to external endpoints for the multicluster observability add-on
+
```
For tech debt, we need to make some changes here:

multicluster observability add-on-- I think we should have coded this?

swopebe commented on observability/export_metrics_ext_endpts.adoc:
```diff
@@ -0,0 +1,48 @@
+
+To configure an external metrics endpoint, add your custom `remoteWrite` specification to your `PrometheusAgents` resources. When you configure your `remoteWrite` specification, the managed cluster sends metrics are directly to the external endpoint.
+
+Update related multicluster observability add-on configurations by relabeling default metrics. Exporting metrics from your managed cluster helps you improve resiliency during network partitions between your managed and hub clusters for up to two hours.
```
this is the stuff that should be early in the topic.
